CMakeLists.txt

```diff
@@ -108,6 +108,12 @@ dgl_feature_option(
     "dev"
 )
 
+dgl_feature_option(
+    USE_ASAN
+    "Use Address Sanitizer(ASAN) to check memory error"
```

Comment:
Can you elaborate more about how to use ASAN?
In this setup, it is none, which means it will never by used? Are you suggesting just turn it on, it will automatically check memory leak and return a report or do we need any tool?

Reply:
It's a cmake option, just turn it on and it'll work. I see you have add a link below introducing ASAN. The reason why I don't set it to "dev" instead of "none" is because there is still a memory leak and if I set it to dev, the UT will fail. please see other comments.

Reply:
Then why do you need to add this CMAKE_CXX_FLAGS in this PR if we add it with OFF?

Reply:
ok, remote it now